src/lib/requestmanager.ts

```diff
@@ -72,23 +95,42 @@ export class RequestManager {
 			this.#clearTimeout = null
 		}
 		const now = Date.now()
-		const qty = this.#entries.size
-		this.#entries.forEach(({ deferred, createdAt }, id) => {
-			if (force || now - createdAt > this.#timeout) {
-				deferred.reject(new Error('ERR_TIMEOUT'))
-				this.#entries.delete(id)
+		const qty = this.#requestsByTime.length
+		// filter() is usually faster than splice() for small-ish arrays
+		this.#requestsByTime = this.#requestsByTime.filter((request) => {
+			if (!this.#requestsById.has(request.invokeId)) {
+				// Request has already been resolved or expired
+				return false
+			}
+			if (force || request.expiresAt <= now) {
+				// Request has timed out or we forcefully time it out
+				request.deferred.reject(new Error('ERR_TIMEOUT'))
+				this.#requestsById.delete(request.invokeId)
+				return false
 			}
+			// Request is still pending
+			return true
 		})
-		debug(`Cleared ${qty - this.#entries.size} entries.`)
-		debug(`There are ${this.#entries.size} entries pending.`)
-		if (this.#entries.size > 0) {
+		assert(
+			this.#requestsById.size === this.#requestsByTime.length,
+			`Index size mismatch  ${this.#requestsById.size} !== ${this.#requestsByTime.length}`,
+		)
```

Comment:
wondering if this is a check you added for tests, or you really want to have this as watchguard?

Reply:
I think we should be 100% sure we don't leak anything instead of causing unexpected client exceptions like this